wallet/wallet.go

```diff
@@ -4,70 +4,358 @@ package wallet
+func (w *LocalWallet) ListAccounts(ctx context.Context) ([]*account.Account, error) {
+	// Return all accounts in the wallet
+	accounts := w.GetAllAccounts()
+	result := make([]*account.Account, len(accounts))
+	for i := range accounts {
+		result[i] = &accounts[i]
+	}
+	return result, nil
+}
+
+// ImportAccount imports an existing Account into the wallet
+func (w *LocalWallet) ImportAccount(ctx context.Context, keyPath string) (*account.Account, error) {
+	// TODO: Implement Account import logic
+	// This would add the provided Account to the wallet
+	existingAccount, err := account.Import(keyPath)
+	if err != nil {
+		return nil, fmt.Errorf("error when importing Account %s \n", err)
+	}
+	w.AddAccount(existingAccount)
+	return &existingAccount, nil
+}
+
+// BuildTx constructs a transaction for the specified operation
+func (w *LocalWallet) BuildTx(ctx context.Context, params BuildTxParams) (tx.BuildTxResult, error) {
+	if err := w.loadAccountIntoWallet(ctx, params.Account, params.Network); err != nil {
+		return tx.BuildTxResult{}, fmt.Errorf("error signing tx: %w", err)
 	}
-	w.options = append(w.options, common.WithChangeOwner(changeOwner))
-	w.Wallet = primary.NewWalletWithOptions(w.Wallet, w.options...)
+	// Validate parameters first
+	if err := params.Validate(); err != nil {
+		return tx.BuildTxResult{}, fmt.Errorf("invalid parameters: %w", err)
+	}
+
+	// Route to appropriate chain handler based on chain type
+	switch chainType := params.GetChainType(); chainType {
+	case "P-Chain":
+		return w.buildPChainTx(ctx, params.Account, params.BuildTxInput)
+	case "C-Chain":
+		return w.buildCChainTx(ctx, params)
+	case "X-Chain":
+		return w.buildXChainTx(ctx, params)
+	default:
+		return tx.BuildTxResult{}, fmt.Errorf("unsupported chain type: %s", chainType)
+	}
+}
+
+// SignTx signs a transaction
+func (w *LocalWallet) SignTx(ctx context.Context, params SignTxParams) (tx.SignTxResult, error) {
+	if err := w.loadAccountIntoWallet(ctx, params.Account, params.Network); err != nil {
+		return tx.SignTxResult{}, fmt.Errorf("error signing tx: %w", err)
+	}
+	if err := w.P().Signer().Sign(context.Background(), params.BuildTxResult.Tx); err != nil {
+		return tx.SignTxResult{}, fmt.Errorf("error signing tx: %w", err)
+	}
+	return tx.SignTxResult{Tx: params.BuildTxResult.Tx}, nil
 }
 
-// SetAuthKeys sets auth keys that will be used when signing txs, besides the wallet's Keychain fee
-// paying ones
-func (w *Wallet) SetAuthKeys(authKeys []ids.ShortID) {
-	addrs := w.Addresses()
-	addrsSet := set.Set[ids.ShortID]{}
-	addrsSet.Add(addrs...)
-	addrsSet.Add(authKeys...)
-	w.options = append(w.options, common.WithCustomAddresses(addrsSet))
-	w.Wallet = primary.NewWalletWithOptions(w.Wallet, w.options...)
+// SendTx submits a signed transaction to the Network
+func (w *LocalWallet) SendTx(ctx context.Context, params SendTxParams) (tx.SendTxResult, error) {
+	if err := w.loadAccountIntoWallet(ctx, params.Account, params.Network); err != nil {
```

Comment:
same here, should be done for different chains

Reply:
Same Q as above.